infinite/index.ts

```diff
@@ -167,7 +167,12 @@ export const infinite = ((<Data, Error, Args extends Arguments>(
 
   const mutate = useCallback(
     (
-      data: Data[] | undefined | Promise<Data[]> | MutatorCallback<Data[]>,
+      data?:
+        | Data[]
+        | Readonly<Data[]>
+        | Promise<Data[]>
+        | Promise<Readonly<Data[]>>
+        | MutatorCallback<Data[]>,
```

Comment:
do we need `MutatorCallback<Readonly<Data[]>>` here?

Reply:
Nope I think, `MutatorCallback<Data = any>` already wraps readonly inside.